zap/src/main/java/org/parosproxy/paros/extension/history/LogPanel.java

```diff
@@ -253,6 +259,35 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
         return filterButton;
     }
 
+    private JButton getClearButton() {
+        if (clearButton == null) {
+            clearButton = new JButton();
+            clearButton.setIcon(
+                    new ImageIcon(
+                            LogPanel.class.getResource(
+                                    "/resource/icon/fugue/broom-alert.png")));
+            clearButton.setToolTipText(
+                    Constant.messages.getString("history.clear.tooltip"));
+            DisplayUtils.scaleIcon(clearButton);
+
+            clearButton.addActionListener(
+                    new java.awt.event.ActionListener() {
+
+                        @Override
+                        public void actionPerformed(java.awt.event.ActionEvent e) {
+                            if (View.getSingleton()
+                                    .showConfirmDialog(
+                                            Constant.messages
+                                                    .getString("history.clear.confirm")) == JOptionPane.OK_OPTION) {
+                                extension.getHistoryReferencesTable().selectAll();
+                                extension.purgeHistory(extension.getSelectedHistoryReferences());
```

Comment:
It would be preferable to add a new method to the table to get all the references instead of selecting (selecting creates events and displays them, which is unnecessary work).

This is a Paros class, they should have a ZAP comment at the top (see existing comments for guidance).

Reply:
> This is a Paros class, they should have a ZAP comment at the top (see existing comments for guidance).

I didn't understand. Which comment at the top?